torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -462,6 +462,13 @@ bool LoopNest::vectorize(For* f) {
   return false;
 }
 
+bool LoopNest::normalizeAndVectorize(For* f) {
+  if (!isNormalized(f)) {
+    normalize(f);
+  }
+  return vectorize(f);
```

Comment:
I'm not sure whether we'll need `vectorize` if `normalizeAndVectorize` is available. The only case `vectorize` works differently from `normalizeAndVectorize` is that it returns false for un-normalized for-loop, which equals to `isNormalized`

Reply:
Or are there cases in which we would not want to perform vectorization if the given loop is not normalized?

Reply:
I thought about having just one function, say normalizing in `vectorize()` method itself directly. The one concern I have with that is if the vectorization fails for some reason, we will still end up normalizing the loop (which I am not sure is the correct contract for a method). So, essentially the method would return `false` but would have still modified the given loop (normalized it).

Reply:
How about we clone the stmt before normalization and try normalize->flatten->vectorize on the cloned stmt, and if any transformation in the chain failed, we just return the original stmt? Would it work?

Reply:
Thats a good idea. In the prior form, we were using the cloned statement even when vectorization fails and I wanted to avoid that. But after my change in this first PR in this stack, we can restrict the cloning and all transformations to the try-catch block. That should work I think. Will make the change.